plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/registry/LanguageServerRegistry.java

```diff
@@ -96,7 +96,7 @@ public void registerFileType(FileType type, LanguageDescription description) {
    * @param file
    * @return
    */
-  public LanguageDescription getLanguageDescription(VirtualFile file) {
+  public LanguageRegex getLanguageFilter(VirtualFile file) {
```

Comment:
This is a `LanguageServerRegistry` so we're talking about languages that has at least one language server associated with. There is no way you can properly configure a language server without defining a regular expression. So you're absolutely right by saying 

>Languages that don't have a language regex, but only a LanguageDescription never get found here.

However that is irrelevant.

Reply:
So we have two independent ways to configure file types: one on the back end via language regex, the other via LanguageDescription. However, since we need a language regex on the back end, we will overwrite part of the configuration done in LanguageDescription anyway.
There can only be one syntax highlighting configuration per editor, but multiple language servers. For me a clear sign that the previous separation of LanguageDescription and LanguageServerDescription was indeed correct.